src/libasr/pass/nested_vars.cpp

```diff
@@ -505,11 +508,19 @@ class AssignNestedVars: public PassUtils::PassVisitor<AssignNestedVars> {
                         LCOMPILERS_ASSERT(sym_ != nullptr);
                         ASR::expr_t *target = ASRUtils::EXPR(ASR::make_Var_t(al, t->base.loc, ext_sym));
                         ASR::expr_t *val = ASRUtils::EXPR(ASR::make_Var_t(al, t->base.loc, sym_));
-                        if( ASRUtils::is_array(ASRUtils::symbol_type(sym)) ||
-                            ASR::is_a<ASR::Pointer_t>(*ASRUtils::symbol_type(sym)) ) {
+                        bool is_sym_allocatable_or_pointer = (ASR::is_a<ASR::Pointer_t>(*ASRUtils::symbol_type(sym)) ||
+                            ASR::is_a<ASR::Allocatable_t>(*ASRUtils::symbol_type(sym)));
+                        bool is_ext_sym_allocatable_or_pointer = (ASR::is_a<ASR::Pointer_t>(*ASRUtils::symbol_type(ext_sym)) ||
+                            ASR::is_a<ASR::Allocatable_t>(*ASRUtils::symbol_type(ext_sym)));
```

Comment:
Shall we reuse `is_pointer() and is_allocatable()` here?
```suggestion
                        bool is_sym_allocatable_or_pointer = (ASRUtils::is_pointer(ASRUtils::symbol_type(sym)) ||
                            ASRUtils::is_allocatable(ASRUtils::symbol_type(sym)));
                        bool is_ext_sym_allocatable_or_pointer = (ASRUtils::is_pointer(ASRUtils::symbol_type(ext_sym)) ||
                            ASRUtils::is_allocatable(ASRUtils::symbol_type(ext_sym)));
```

Reply:
I think it can be done.